src/components/explorer/hooks/explorerHooks.js

```diff
@@ -29,3 +29,13 @@ export const useSortedColumns = (data, tableSortOrder, columnsDefinition) => {
 
     return { sortedData, columnsWithSortOrder };
 };
+
+export const useDynamicFilterOptions = (data, key) => {
```

Comment:
mention in hook name that this is for tables, like `useDynamicTableFilterOptions`